backends/tfhe-cuda-backend/cuda/src/integer/compression/compression.cuh

```diff
@@ -283,88 +295,111 @@ host_integer_decompress(cudaStream_t const *streams,
+  Torus *extracted_lwe;
+  if constexpr (std::is_same_v<Torus, uint64_t>) {
+    // 64 bits
+    extracted_lwe = h_mem_ptr->tmp_extracted_lwe;
+  } else {
+    // 128 bits
+    // We skip the PBS at the end
+    extracted_lwe = static_cast<Torus *>(d_lwe_array_out->ptr);
+  }
   uint32_t current_idx = 0;
   auto d_indexes_array_chunk = d_indexes_array;
   for (const auto &max_idx_and_glwe : glwe_vec) {
     const auto num_lwes = max_idx_and_glwe.first;
     extracted_glwe = max_idx_and_glwe.second;
 
-    cuda_glwe_sample_extract_64(
-        streams[0], gpu_indexes[0], extracted_lwe, extracted_glwe,
-        d_indexes_array_chunk, num_lwes, compression_params.polynomial_size,
-        compression_params.glwe_dimension, compression_params.polynomial_size);
+    if constexpr (std::is_same_v<Torus, uint64_t>)
+      cuda_glwe_sample_extract_64(streams[0], gpu_indexes[0], extracted_lwe,
+                                  extracted_glwe, d_indexes_array_chunk,
+                                  num_lwes, compression_params.polynomial_size,
+                                  compression_params.glwe_dimension,
+                                  compression_params.polynomial_size);
+    else
+      // 128 bits
+      cuda_glwe_sample_extract_128(streams[0], gpu_indexes[0], extracted_lwe,
+                                   extracted_glwe, d_indexes_array_chunk,
+                                   num_lwes, compression_params.polynomial_size,
+                                   compression_params.glwe_dimension,
+                                   compression_params.polynomial_size);
+
     d_indexes_array_chunk += num_lwes;
-    extracted_lwe += num_lwes * lwe_accumulator_size;
+    extracted_lwe += num_lwes * lwe_size;
     current_idx += num_lwes;
   }
 
-  // Reset
-  extracted_lwe = h_mem_ptr->tmp_extracted_lwe;
-
-  // In the case of extracting a single LWE these parameters are dummy
-  uint32_t num_many_lut = 1;
-  uint32_t lut_stride = 0;
-  /// Apply PBS to apply a LUT, reduce the noise and go from a small LWE
-  /// dimension to a big LWE dimension
-  auto encryption_params = h_mem_ptr->encryption_params;
-  auto lut = h_mem_ptr->decompression_rescale_lut;
-  auto active_gpu_count =
-      get_active_gpu_count(num_blocks_to_decompress, gpu_count);
-  if (active_gpu_count == 1) {
-    execute_pbs_async<Torus>(
-        streams, gpu_indexes, active_gpu_count, (Torus *)d_lwe_array_out->ptr,
-        lut->lwe_indexes_out, lut->lut_vec, lut->lut_indexes_vec, extracted_lwe,
-        lut->lwe_indexes_in, d_bsks, nullptr, lut->buffer,
-        encryption_params.glwe_dimension,
-        compression_params.small_lwe_dimension,
-        encryption_params.polynomial_size, encryption_params.pbs_base_log,
-        encryption_params.pbs_level, encryption_params.grouping_factor,
-        num_blocks_to_decompress, encryption_params.pbs_type, num_many_lut,
-        lut_stride);
+  if constexpr (std::is_same_v<Torus, uint64_t>) {
```

Comment:
Same here for the if/else, we should check explicitely it's uint128 and if not PANIC